src/code/audio/code_8019AF00.c

```diff
@@ -4463,35 +4744,238 @@ void Audio_ClearSariaBgm2(void) {
+    u16 seqId;
+    u8 volumeFadeOutTimer;
+
+    if ((sPrevMainBgmSeqId == NA_BGM_DISABLED) && (sPrevMainBgmSeqId == NA_BGM_DISABLED)) {
+        // clang-format off
+        if (sAudioCutsceneFlag || sSpatialSeqIsActive[SEQ_PLAYER_BGM_SUB]) { \
+            seqMode = SEQ_MODE_IGNORE;
+        }
+        // clang-format on
+
+        seqId = gActiveSeqs[SEQ_PLAYER_BGM_MAIN].seqId;
+
+        if ((seqId == NA_BGM_DISABLED) || (sSeqFlags[(u8)(seqId & 0xFF)] & SEQ_FLAG_ENEMY) ||
+            ((sPrevSeqMode & 0x7F) == SEQ_MODE_ENEMY)) {
+            if (seqMode != (sPrevSeqMode & 0x7F)) {
+                if (seqMode == SEQ_MODE_ENEMY) {
+                    // If only seqMode = SEQ_MODE_ENEMY (Start)
+                    if (gActiveSeqs[SEQ_PLAYER_BGM_SUB].volScales[1] - sBgmEnemyVolume < 0) {
+                        volumeFadeInTimer = -(gActiveSeqs[SEQ_PLAYER_BGM_SUB].volScales[1] - sBgmEnemyVolume);
+                    } else {
+                        volumeFadeInTimer = gActiveSeqs[SEQ_PLAYER_BGM_SUB].volScales[1] - sBgmEnemyVolume;
+                    }
+
+                    Audio_SetVolumeScale(SEQ_PLAYER_BGM_SUB, 3, sBgmEnemyVolume, volumeFadeInTimer);
+                    Audio_QueueSeqCmd(0x00000000 | ((u32)(SEQ_PLAYER_BGM_SUB) << 24) | ((u32)(10) << 0x10) |
+                                      (u32)(NA_BGM_ENEMY | 0x800));
+
+                    if (seqId >= NA_BGM_TERMINA_FIELD) {
+                        Audio_SetVolumeScale(SEQ_PLAYER_BGM_MAIN, 3, 0x7F - sBgmEnemyVolume, 10);
+                        Audio_SplitBgmChannels(sBgmEnemyVolume);
+                    }
+                } else if ((sPrevSeqMode & 0x7F) == SEQ_MODE_ENEMY) {
+                    // If only sPrevSeqMode = SEQ_MODE_ENEMY (End)
+                    Audio_QueueSeqCmd(0x100000FF | ((u8)(SEQ_PLAYER_BGM_SUB) << 24) | ((10) << 16));
+
+                    if (seqMode == SEQ_MODE_IGNORE) {
+                        volumeFadeOutTimer = 0;
+                    } else {
+                        volumeFadeOutTimer = 10;
+                    }
+
+                    Audio_SetVolumeScale(SEQ_PLAYER_BGM_MAIN, 3, 0x7F, volumeFadeOutTimer);
+                    Audio_SplitBgmChannels(0);
+                }
+
+                sPrevSeqMode = seqMode + 0x80;
+            } else {
+                if (seqMode == SEQ_MODE_ENEMY) {
+                    // If both seqMode = sPrevSeqMode = SEQ_MODE_ENEMY
+                    if ((Audio_GetActiveSequence(SEQ_PLAYER_BGM_SUB) == NA_BGM_DISABLED) &&
+                        (seqId != NA_BGM_DISABLED) && (sSeqFlags[seqId & 0xFF & 0xFF] & SEQ_FLAG_ENEMY)) {
+                        Audio_QueueSeqCmd(0x00000000 | ((u32)(SEQ_PLAYER_BGM_SUB) << 24) | ((u32)(10) << 0x10) |
+                                          (u32)(NA_BGM_ENEMY | 0x800));
+                        sPrevSeqMode = seqMode + 0x80;
+                    }
+                }
+            }
+        } else {
+            // Remnant of OoT's Hyrule Field Sequence
```

Comment:
Oh, I suppose that TF is nowhere near as sophisticated?

Reply:
It doesn't appear so, no. Would need seq extraction to 100% confirm but it doesn't seem so from what I tested